pkg/cloud/services/networking/port.go

```diff
@@ -192,7 +192,7 @@ func (s *Service) CreatePort(eventObject runtime.Object, portSpec *infrav1.Resol
 
 	port, err := s.client.CreatePort(builder)
 	if err != nil {
-		record.Warnf(eventObject, "FailedCreatePort", "Failed to create port %s: %v", port.Name, err)
+		record.Warnf(eventObject, "FailedCreatePort", "Failed to create port %s: %v", portSpec.Name, err)
```

Comment:
I'm fine with this change because:

* `port` is indeed a pointer so if `CreatePort` returns nil, we can't use it and we'll have to check for it here before.
* We already use `portSpec.Name` in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/349ec37fd8b3ed15a6750f4b3506993d8d287cc4/pkg/cloud/services/networking/port.go#L201-L202